include/benchmark/benchmark_api.h

```diff
@@ -247,34 +247,39 @@ enum BigO {
   oNCubed,
   oLogN,
   oNLogN,
-  oAuto
+  oAuto,
+  oLambda
 };
 
+// BigOFunc is passed to a benchmark in order to specify the asymptotic 
+// computational complexity for the benchmark.
+typedef double(BigOFunc)(size_t);
```

Comment:
No, it doesn't compile on the c++0x compiler because it generates the error _"function’ in namespace ‘std’ does not name a template type_ I investigated the problem, and I found out that this happens with versions of g++ less that 4.7 and we have target compilers that are g++4.6. 
In fact, the first implementation I did was with std::function<double(size_t)> but I had to change it by the function definition to make it compile on target compilers.